src/ripple/overlay/impl/TrafficCount.h

```diff
@@ -178,50 +178,51 @@ class TrafficCount
+        {"shards"},                                          // category::shards
+        {"set_get"},                                         // category::get_set
+        {"set_share"},                                       // category::share_set
+        {"ledger_data_Transaction_Set_candidate_get"},       // category::ld_tsc_get
+        {"ledger_data_Transaction_Set_candidate_share"},     // category::ld_tsc_share
+        {"ledger_data_Transaction_Node_get"},                // category::ld_txn_get
+        {"ledger_data_Transaction_Node_share"},              // category::ld_txn_share
+        {"ledger_data_Account_State_Node_get"},              // category::ld_asn_get
+        {"ledger_data_Account_State_Node_share"},            // category::ld_asn_share
+        {"}ledger_data_get"},                                 // category::ld_get
+        {"ledger_data_share"},                               // category::ld_share
+        {"ledger_Transaction_Set_candidate_share"},          // category::gl_tsc_share
+        {"ledger_Transaction_Set_candidate_get"},            // category::gl_tsc_get
+        {"ledger_Transaction_node_share"},                   // category::gl_txn_share
+        {"ledger_Transaction_node_get"},                     // category::gl_txn_get
+        {"ledger_Account_State_node_share"},                 // category::gl_asn_share
+        {"ledger_Account_State_node_get"},                   // category::gl_asn_get
+        {"ledger_share"},                                    // category::gl_share
+        {"ledger_get"},                                      // category::gl_get
+        {"getobject_Ledger_share"},                          // category::share_hash_ledger
+        {"getobject_Ledger_get"},                            // category::get_hash_ledger
+        {"getobject_Transaction_share"},                     // category::share_hash_tx
+        {"getobject_Transaction_get"},                       // category::get_hash_tx
+        {"getobject_Transaction_node_share"},                // category::share_hash_txnode
+        {"getobject_Transaction_node_get"},                  // category::get_hash_txnode
+        {"getobject_Account_State_node_share"},              // category::share_hash_asnode
+        {"getobject_Account_State_node_get"},                // category::get_hash_asnode
+        {"getobject_CAS_share"},                             // category::share_cas_object
+        {"getobject_CAS_get"},                               // category::get_cas_object
+        {"getobject_Fetch_Pack_share"},                      // category::share_fetch_pack
+        {"getobject_Fetch Pack_get"},                        // category::get_fetch_pack
+        {"getobject_share"},                                 // category::share_hash
+        {"getobject_get"},                                   // category::get_hash
+        {"unknown"}                                          // category::unknown
```

Comment:
Storing these names in the `TrafficStats` object is causing trouble: it has two constructors, neither of which is default, and there's a `std::string` being hauled around in every `TrafficStats` object, wasting space. I searched and no one is ever passing around a single `TrafficStats` object. They are written by their index (via the enum `category`), and only ever read while iterating through the whole collection.

If that is true, then we could pull the name out of `TrafficStats`, and use the index during iteration to access a `constexpr array` of the names, stored as `char const *` instead of `std::string`. The `TrafficStats` class can be given a default constructor. The `counts_` array can delegate to the default constructor by just initializing it with the size.

At the very least, the `name` member of `TrafficStats` can be changed to `char const *`.

Reply:
Thanks, I changed the `name` member to a `char const* `